lib/datadog/statsd.rb

```diff
@@ -441,14 +478,14 @@ def send_stat(message)
         @buffer << message
         flush_buffer if @buffer.length >= @max_buffer_size
       else
-        send_to_socket(message)
+        send_to_socket_synchronized(message)
       end
     end
 
     def flush_buffer
       return @buffer if @buffer.empty?
-      send_to_socket(@buffer.join(NEW_LINE))
-      @buffer = Array.new
+      send_to_socket_synchronized(@buffer.join(NEW_LINE))
+      @buffer = Concurrent::Array.new
```

Comment:
can use synchronize here too and a regular array ?

Reply:
true and legit if you want to keep the dependency footprint low. Although Concurrent::Array does all of that for us.